src/keyring/KeyringController.test.ts

```diff
@@ -469,11 +628,56 @@ describe('KeyringController', () => {
     expect(signedTx).not.toBe('');
   });
 
+  it('should not sign transaction if from account is not provided', async () => {
+    await expect(async () => {
+      const account = initialState.keyrings[0].accounts[0];
+      const txParams = {
+        chainId: 3,
+        data: '0x1',
+        from: account,
+        gasLimit: '0x5108',
+        gasPrice: '0x5108',
+        to: '0x51253087e6f8358b5f10c0a94315d69db3357859',
+        value: '0x5208',
+      };
+      const unsignedEthTx = TransactionFactory.fromTxData(txParams, {
+        common: new Common(commonConfig),
+        freeze: false,
+      });
+      expect(unsignedEthTx.v).toBeUndefined();
+      await keyringController.signTransaction(unsignedEthTx, '');
+    }).rejects.toThrow(
+      'No keyring found for the requested account. Error info: The address passed in is invalid/empty; There are keyrings, but none match the address;',
+    );
+  });
+
+  /**
+   * Task added to improve check for valid transaction in signTransaction method
+   * https://github.com/MetaMask/controllers/issues/800
+   */
+  it('should not sign transaction if transaction is not valid', async () => {
+    await expect(async () => {
+      const account = initialState.keyrings[0].accounts[0];
+      const signedTx = await keyringController.signTransaction({}, account);
+      expect(signedTx.v).not.toBeUndefined();
+      expect(signedTx).not.toBe('');
```

Comment:
[nit] I think you can drop these `expect`s, yeah? Because it should error before this point.

Reply:
Good point fixed.